supercollider/classes/PirateRadio.sc

```diff
@@ -448,10 +439,44 @@ PradStreamPlayer {
 			numChannels=l2.asInteger;
 			numFrames=l3.asFloat;
 			durationSeconds=l.asFloat;
-			if (startSeconds<durationSeconds,{
-				durationSeconds=durationSeconds-startSeconds;
-			});
 		});
+
+		Array.with(durationSeconds, numChannels, numFrames);
+	}
+
+	playFile {
+		arg fname, startSeconds=0;
+		var pr;
+		var durationSeconds=1,numChannels=2,numFrames=1.0;
+		var xfade=0;
+		var currentFileScheduler=0;
+		var originalFname=fname;
+
+		// swap synths/buffers
+		this.postln;
+		("station "++id++" playing file "++fname.asAbsolutePath).postln;
+		swap=1-swap;
+		fnames[swap]=(fname.asAbsolutePath).asString;
+
+		// send update to server that a song is playing
+		NetAddr("127.0.0.1", 10111).sendMsg("playing",id,fnames[swap]);
+
+		if (ffprobes.includesKey(fname.asAbsolutePath),{
+			pr=ffprobes[fname.asAbsolutePath];
+		},{
+			pr=this.probe(fname.asAbsolutePath);
```

Comment:
This doesn't work, it assigns the object (`this`) to `pr` instead of the result of the method call. Why?

Reply:
it might be a mistake